### PR TITLE
[Dashboard] Fix discrepancies for "Violated Scans" in Dashboard

### LINKED ISSUE #5289: [Dashboard] Investigate discrepancies for total displayed for "Incomplete Forms" and "Violated Scans" in Dashboard
Please investigate discrepancies found in dashboard display for Incomplete Forms and MRI Violation.

Violated scans - Total displayed 10,828. Total in MRI Violated Scans module = 16,834, minus 5,953 for phantom scans, minus 32 for non-phantom DCC entries = 10,859. Unsure of where difference of 31 scans is coming from. 

Migrated from: https://redmine.cbrain.mcgill.ca/issues/14106

### PR DESCRIPTION
**Pull request by sruthymathew123**
## Brief summary of changes
-Fixes #5289 (MRI Violations Discrepancies)

#### Testing instructions (if applicable)

1. Before checking out this PR, make a note on the count on mri violated count on dashboard and as well as in mri violations module. (There will be mismatch in the count)
2. Checkout to this branch and see if the MRI violation count in Dashboard matches with Unresolved violations in MRI violations module,

NOTE: The database query used to fetch mri violations log in mri violations module is  moved to a class constant and referred to in  `getWidgets()` function to fix the discrepancies.
I suggest this to be tested in project having an actual imaging dataset.

### CODE REVIEW

File: modules/mri_violations/php/module.class.inc [hunk 1]
```diff
@@ -85,9 +85,93 @@ class Module extends \Module
                     $user,
                     "Violated scan",
                     $DB,
-                    "SELECT COUNT(*) FROM mri_protocol_violated_scans
-                        LEFT JOIN candidate c USING (CandID)
-                    WHERE 1 = 1 ",
+                    "SELECT
```
Comment (driusan) [inline modules/mri_violations/php/module.class.inc:88]: If this is the exact same query (and it's this long..) maybe it should go into a class constant and just be referred to in both places?

Comment (sruthymathew123) [reply modules/mri_violations/php/module.class.inc:88]: @driusan I would say almost exact (but ignored some columns). I actually don't have any idea if there is any way to make the query small. It is the only way I could find make the count to be exact in Dashboard.

Comment (driusan) [reply modules/mri_violations/php/module.class.inc:88]: but can you move it into a constant and just refer to the constant in both places?

File: modules/mri_violations/php/module.class.inc [hunk 2]
```diff
@@ -85,9 +187,8 @@ class Module extends \Module
...
+               count(*)".self::MRI_VIOLATED_SCAN_QUERY_CONST,
```
Comment (lingma) [inline modules/mri_violations/php/module.class.inc:191]: Here it should be `count(PatientName)`

File: modules/mri_violations/php/module.class.inc (hunk 1 above)
Comment (sruthymathew123) [reply modules/mri_violations/php/module.class.inc:190]: @driusan changes Done as per your suggestion. Let me know if it is all good now.

Comment (driusan) [reply modules/mri_violations/php/module.class.inc:190]: Looks fine now, but it has conflicts and can't be merged.

Comment (sruthymathew123) [reply modules/mri_violations/php/module.class.inc:190]: @driusan Oh, ok. just noticed that. I will fix it.

Comment (sruthymathew123) [reply modules/mri_violations/php/module.class.inc:199]: @driusan Done. Fixed the conflicts.

**Pull request merged**